src/common/ServerSocket.cpp

```diff
@@ -0,0 +1,43 @@
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <unistd.h>
+
+#include "ServerSocket.h"
+#include "../Exceptions.h"
+#include "Utils.h"
+
+ServerSocket::ServerSocket(int socket_fd)
+    : m_socket_fd(socket_fd)
+{
+}
+
+ServerSocket::~ServerSocket()
+{
+    ::close(m_socket_fd);
```

Comment:
u use this function twice, both server_socket and socket use close at the dtor, consider combining them into a single RAII wrapper (maybe smart guard)